tls/extensions/s2n_client_key_share.c

```diff
@@ -175,7 +175,17 @@ static int s2n_generate_default_pq_hybrid_key_share(struct s2n_connection *conn,
          **/
         client_params->kem_group = server_group;
     } else {
-        client_params->kem_group = kem_pref->tls13_kem_groups[0];
+        client_params->kem_group = NULL;
+        for (int i = 0; i < kem_pref->tls13_kem_group_count; i++) {
+            if (kem_pref->tls13_kem_groups[i]->available) {
+                client_params->kem_group = kem_pref->tls13_kem_groups[i];
+                break;
+            }
+        }
+        /* Ignore unavailable KEM groups */
+        if (client_params->kem_group == NULL) {
+            return S2N_SUCCESS;
+        }
```

Comment:
I'm not sure it's possible to hit this case since `s2n_kem_groups_available_count` is checked above. Should this maybe just ENSURE_REF?
```suggestion
        POSIX_ENSURE_REF(client_params->kem_group);
```

Reply:
Was this change missed?

Reply:
whoops, yeah I missed this. fixed.